### PR TITLE
App-wide configurations for enterprise and masters

### PR DESCRIPTION
**Pull request by adamstankiewicz**
My proposed solution for [ENT-2280](https://openedx.atlassian.net/browse/ENT-2280) involves some refactoring to add 3 new components: `AppContext`, `EnterprisePage`, and `MastersPage`. It also removes `LayoutContext` in favor of `AppContext`.

Now, `Layout` (and other components) pull from `AppContext` to get whatever configuration it needs. The value for `AppContext` is set in `EnterprisePage` and `MastersPage`. The value is meant to be an app-wide configuration, where you can pass in the `pageContext` as well as header menu items that should only apply for masters vs. enterprise, without needing to do any conditionals based on `pageType` (e.g., "pages.EnterprisePage").

This also tackles the TODO [noted here](https://github.com/edx/frontend-app-learner-portal/blob/c748b14ce06ef1c774c931440a2def0a72db048f/src/components/common/course-enrollments/course-cards/InProgressCourseCard.jsx#L53). Because this app-wide configuration could also include configurations/overrides for features, such as whether the "Mark as complete" button appears for enterprise.

Example value for `AppContext` (e.g., set in `EnterprisePage`):
```
{
  header: {
    mainMenu: [
      {
        type: 'item',
        href: process.env.ENTERPRISE_CATALOG_MFE_URL,
        content: 'Catalog',
      },
      {
        type: 'item',
        href: 'https://support.edx.org/hc/en-us',
        content: 'Help',
      },
    ],
    userMenu: [
      {
        type: 'item',
        href: process.env.LMS_BASE_URL,
        content: 'Dashboard',
      },
      {
        type: 'item',
        href: `${process.env.LMS_BASE_URL}/u/${username}`,
        content: 'My Profile',
      },
      // ...
    ],
  },
  courseCards: {
    'in-progress': {
      settingsMenu: {
        hasMarkComplete: true,
      },
    },
  },
  pageContext,
}
```
I feel like this sort of approach could make sense, but curious to hear any feedback. In situations like whether the "Mark as complete" button shows, we still have a conditional to see if `hasMarkComplete` exists & is true. But, at least, it's a conditional that isn't explicitly dependent on a `pageType`.

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -40,6 +40,7 @@ module.exports = {
         whitelist: [
           'BASE_URL',
           'ENTERPRISE_CATALOG_MFE_URL',
+          'ORDERS_MFE_URL',
```
Comment (adamstankiewicz) [inline gatsby-config.js]: For the "Order History" link in the user dropdown menu in the header.

File: src/components/common/course-enrollments/CourseEnrollments.jsx [hunk 2]
```diff
@@ -19,21 +19,20 @@ import * as selectors from './data/selectors';
...
-        pageType,
         programUUID, // for Masters, empty for Enterprise
         enterpriseUUID, // for Enterprise, empty for Masters
       },
     } = this.context;
     const { fetchCourseEnrollments } = this.props;
-    const options = { pageType };
-    if (pageType === 'pages.ProgramPage') {
+    const options = {};
+    if (programUUID) {
```
Comment (adamstankiewicz) [inline src/components/common/course-enrollments/CourseEnrollments.jsx]: It's not really necessary to check based on `pageType` here. Instead, we can rely on the existence of `programUUID` and `enterpriseUUID` variables.

File: src/components/masters/masters-page/MastersPage.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+const MastersPage = ({
+  children,
+  pageContext,
+  username,
+}) => (
+  <AppContext.Provider
+    value={{
+      header: {
+        userMenu: [
```
Comment (adamstankiewicz) [inline src/components/masters/masters-page/MastersPage.jsx]: Passing all the header menu items in this sort of config gives more control to masters & enterprise to control their own individual menu options, including being able to order the links as needed.

File: .env.development [hunk 4]
```diff
@@ -1,5 +1,6 @@
 BASE_URL='localhost:8734'
-ENTERPRISE_CATALOG_MFE_URL=''
+ENTERPRISE_CATALOG_MFE_URL='http://example.com'
+ORDERS_MFE_URL='http://localhost:1996'
```
Comment (adamstankiewicz) [inline .env.development]: `ORDERS_MFE_URL` for development is the port that the frontend-app-ecommerce MFE runs on. Note: If this is the right approach for how we manage what that URL is, we'll need to add this environment variable to edx-internal so it's part of stage/prod builds.

File: src/components/masters/masters-page/MastersPage.jsx (hunk 3 above)
Comment (christopappas) [reply src/components/masters/masters-page/MastersPage.jsx]: 👍

**Pull request merged**